config/intellij-idea-inspections.xml

```diff
@@ -105,9 +105,14 @@
     <inspection_tool class="AssignmentUsedAsCondition" enabled="true" level="ERROR" enabled_by_default="true" />
     <!-- we are ok to use auto-boxing as we use modern java -->
     <inspection_tool class="AutoBoxing" enabled="false" level="ERROR" enabled_by_default="false" />
-    <!-- we can not use try-with-resource till we switch to jacoco for code coverage -->
-    <inspection_tool class="AutoCloseableResource" enabled="false" level="ERROR" enabled_by_default="false" />
-    <!-- we are ok to use auto-unboxing as we use modern java -->
+    <inspection_tool class="AutoCloseableResource" enabled="true" level="ERROR" enabled_by_default="true">
+        <!-- These classes do not contain any resources that need to be closed. -->
+        <option name="ignoredTypes" value="java.io.ByteArrayOutputStream,java.util.stream.Stream,java.util.stream.IntStream,java.util.stream.LongStream,java.util.stream.DoubleStream,com.puppycrawl.tools.checkstyle.internal.utils.CloseAndFlushTestByteArrayOutputStream" />
+        <!-- False positives for the Closable returned from a builder method (which return "this").
+             Mock stubs shouldn't be closed. -->
+        <option name="METHOD_MATCHER_CONFIG" value="java.util.Formatter,format,java.io.Writer,append,com.google.common.base.Preconditions,checkNotNull,org.hibernate.Session,close,java.io.PrintWriter,printf,org.mockito.ArgumentCaptor,getValue,org.mockito.Mockito,mock|verify,org.mockito.stubbing.Stubber,when" />
```

Comment:
> getValue

it is not ok. We can resolve this bad code.

please review commits, sorry if they are messy, I can not squash in main repo:
https://github.com/checkstyle/checkstyle/commit/2b0ac1b899d0ff3dbde197b2bb1c9c579663f80f
https://github.com/checkstyle/checkstyle/commit/9c6aa47b5f3eb2307ab005983bbd3b1d43a9fedc
I refactored createListener to create streams when required `getOutputStream(outputLocation)` and `getOutputStreamOptions(String outputLocation)`.

If it still messy, I will recreate branch.
If you have some concerns with such implementation, we can merge this PR, but discuss update in separate issue.